flink-core/src/main/java/org/apache/flink/configuration/Configuration.java

```diff
@@ -765,6 +765,18 @@ public <T> boolean removeConfig(ConfigOption<T> configOption) {
         }
     }
 
+    /**
+     * Removes given key from the configuration.
+     *
+     * @param key key of a config option to remove
+     * @return true is config has been removed, false otherwise
+     */
+    public boolean removeKey(String key) {
+        synchronized (this.confData) {
+            return this.confData.remove(key) != null;
```

Comment:
Let's also call `removePrefixMap` here to make sure all variations of this key are gone. This should be safe as all keys represent leaves.